src/provider/store/Actions.ts

```diff
@@ -13,47 +14,91 @@ export const enum Action {
     TOGGLE_VISIBILITY = '@@ui/TOGGLE_CENTER_WINDOW',
 }
 
-export interface CreateNotification extends ReduxAction<Action> {
-    type: Action.CREATE,
-    notification: StoredNotification;
+export class BaseAction<T extends Action> implements ReduxAction<Action> {
+    public readonly type: T;
+
+    constructor(type: T) {
+        this.type = type;
+    }
+}
+
+export abstract class CustomAction<T extends Action> extends BaseAction<T> {
+    public abstract async dispatch(store: StoreAPI): Promise<void>;
+}
+
+export class CreateNotification extends CustomAction<Action.CREATE> {
+    public readonly notification: StoredNotification;
+
+    constructor(notification: StoredNotification) {
+        super(Action.CREATE);
+        this.notification = notification;
+    }
+
+    public async dispatch(store: StoreAPI): Promise<void> {
+        const notification = this.notification;
+        const existingNotifications = store.state.notifications.filter(x => x.id === notification.id);
```

Comment:
Something's gone very wrong if we've got two notifications with the same ID, right? Would go with `find` rather than `filter`, so as to not miscommunicate that we reasonably expect there to be multiple notifs with the same ID. And then if we want to check if we've accidentally stored notifs with duplicate IDs, we do that explicitly (at service startup?)

Reply:
As above, will persist but flag.

In this case, the Action only accepts an array of notifications, so would only end up wrapping the single notification in an array anyway.